### PR TITLE
[9.x] Add Model::withoutTimestamps(...)

### PR DESCRIPTION
**Pull request by timacdonald**
Re-submission of #44005 as static methods.

API now matching the static nature of `withoutTouching`, but hopefully with a simpler implementation than previous attempts at this functionality.

```php
$user = User::first();


// `updated_at` is not changed...

User::withoutTimestamps(
    fn () => $user->update(['reserved_at' => now()])
);
```

### CODE REVIEW

File: src/Illuminate/Database/Eloquent/Concerns/HasTimestamps.php [hunk 1]
```diff
@@ -113,7 +120,7 @@ public function freshTimestampString()
      */
     public function usesTimestamps()
     {
-        return $this->timestamps;
+        return $this->timestamps && ! static::isIgnoringTimestamps($this::class);
```
Comment (wotta) [inline src/Illuminate/Database/Eloquent/Concerns/HasTimestamps.php:123]: According to all other parts of you PR shouldn't you use `static::class` here?

**Comment by marcovo** [comment]
I suspect this implementation will not work as expected with nested calls to `withoutTimestamps()`. E.g.:

```
User::withoutTimestamps(function () use ($user) {

    User::withoutTimestamps(
        fn () => $user->update(['reserved_at' => now()])
    );
    
    $user->update([...]); // <-- Will update timestamps
});
```

You can imagine one would call `withoutTimestamps()` e.g. in a controller and then deeper somewhere in a model, in different contexts, which would lead to this situation. I think it should be fixed to be able to cope with this situation.

**Comment by timacdonald** [comment]
I did not intend for it to be used this way. Would recommend not nesting calls to this or the other methods that work this way.

**Comment by jasonmccreary** [comment]
Yeah, I've run into this before. Disabling timestamps is a tricky beast and heavily dependent on the use case.

**Comment by moisish** [comment]
I had my own implementation of withoutTimestamps on trait that was working differently. 
The withoutTimestamps was a non static method that I was able to use on the existing model I am using or on an Eloquent Builder instance

On a single model
```
$model
    ->withoutTimestamps()
    ->update([
           'included' => true
      ]);
```

Updating all models by using a relationship
```
$tasks
            ->subTasks()
            ->tap(function (Builder $query) {
                $query->getModel()->withoutTimestamps();
            })
            ->update([
                'included' => false,
            ]);
```

**Comment by dennisprudlo** [comment]
Wouldn't both example be possible like this?
```php
Model::withoutTimestamps(fn () => $model->update([ 'included' => true ]));
```

```php
SubTask::withoutTimestamps(fn () => $task->subTasks()->update([ 'included' => false ]));
```

**Comment by timacdonald** [comment]
See @dennisprudlo suggestion above, and also see the previous PR where the method was not static for future discussion: https://github.com/laravel/framework/pull/44005

**Pull request merged**